sections/implementation/frontend-structure.tex

```diff
@@ -6,17 +6,16 @@ \subsection{States}
 For many of our pages we use states to store a component's dynamic data. 
 An example of this is the edit service page. 
 The constructor for this page is seen on \autoref{EditService-state}.
-Within the constructor \texttt{super(props)} is called. 
-This is used to call the React parent constructor and pass its properties as an argument.
-This is necessary whenever you want to call \texttt{this} within the constructor.
-Variables that will be used on the page are found within \texttt{this.state}. 
+The constructor takes a variable called props as an argument, props contains all of the arguments that the component received when it was instantiated. 
+Within the constructor \texttt{super(props)} is called.
+This needs to be done to be able to use \texttt{this.props} to access the props anywhere in the component.
+Dynamic variables that will be used on the page are initialized in \texttt{this.state}. 
 However, if another variable is needed, it can be added to the state at any time as the state is dynamic.
-The page will get rendered again whenever the state changes. 
-This is very useful for many things. 
+Variables in the state can be changed by calling the react hook \texttt{this.setState()} and this will automatically trigger a re rendering of the component and all of it's children.
```

Comment:
```suggestion
Variables in the state can be changed by calling the react hook \texttt{this.setState()} and this will automatically trigger a new rendering of the component and all of its children.
```